xtask/src/main.rs

```diff
@@ -41,6 +43,11 @@ fn main() -> xshell::Result<()> {
     {
         let _s = section("TEST_BETA");
         let _e = push_toolchain(&sh, "beta")?;
+        // TEMPORARY WORKAROUND for Rust compiler issue ref:
+        // - https://github.com/rust-lang/rust/issues/129352
+        // - https://github.com/matklad/once_cell/issues/261
+        let _e = sh.push_env("RUSTFLAGS", "-A unreachable_patterns");
```

Comment:
Ahh, I didn't realize this percolated all the way up to the beta! But, anyway, you've found the _perfect_ solution here, which places the fine-grained work-around where it belongs, thanks a bunch for this!